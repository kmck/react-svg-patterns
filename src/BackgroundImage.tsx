import * as React from 'react';

interface Props {
  id: string,
  src?: string
};

const BackgroundImage = ({
  id,
  src = ''
}: Props) => (
  <pattern
    height={1}
    id={id}
    patternContentUnits="objectBoundingBox"
    patternUnits="objectBoundingBox"
    width={1}
  >
    <image
      height={1}
      width={1}
      x={0}
      xlinkHref={src}
      y={0}
    />
  </pattern>
);

export default React.memo(BackgroundImage);
