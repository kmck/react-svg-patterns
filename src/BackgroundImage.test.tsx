import * as React from 'react';
import renderer from 'react-test-renderer';

import BackgroundImage from './BackgroundImage';

describe('BackgroundImage', () => {
  it('renders a background image pattern', () => {
    const tree = renderer.create(
      <BackgroundImage id="test" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a background image pattern with the specified src', () => {
    const tree = renderer.create(
      <BackgroundImage id="test" src="https://via.placeholder.com/150" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
