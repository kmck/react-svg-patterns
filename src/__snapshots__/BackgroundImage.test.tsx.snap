// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage renders a background image pattern 1`] = `
<pattern
  height={1}
  id="test"
  patternContentUnits="objectBoundingBox"
  patternUnits="objectBoundingBox"
  width={1}
>
  <image
    height={1}
    width={1}
    x={0}
    xlinkHref=""
    y={0}
  />
</pattern>
`;

exports[`BackgroundImage renders a background image pattern with the specified src 1`] = `
<pattern
  height={1}
  id="test"
  patternContentUnits="objectBoundingBox"
  patternUnits="objectBoundingBox"
  width={1}
>
  <image
    height={1}
    width={1}
    x={0}
    xlinkHref="https://via.placeholder.com/150"
    y={0}
  />
</pattern>
`;
